fix: vector store initialization and handling

pakagronglb opened the pull request:
- Ensure proper vector store creation and persistence
- Add error handling for vector store operations
- Improve logging for vector store status

Please let me know if it works on your end

I use this dummy literature to see if it works. Literally called Vectors lol
[Vectors Addition and Subtraction.pdf](https://github.com/user-attachments/files/18584123/Vectors.Addition.and.Subtraction.pdf)

TheCarBun commented on app.py:
I tried this out and it successfully displays error message when it can't find any text in the PDF

![image](https://github.com/user-attachments/assets/09a1df57-8232-4314-b32a-eac48d3e07b6)

TheCarBun reviewed (changes requested):
The only few changes that I'd like is:
- to show only one error message if possible.
- diable the text input until the user removes the pdf or remove the pdf and show error messages as toast messages using `st.toast` so it vanishes after few seconds and the user can reupload another pdf like a fresh start.

pakagronglb commented on app.py:
```diff
@@ -171,13 +175,17 @@ def get_vectorstore():
         st.rerun()
       except Exception as e:
-        st.error(f"Error creating vectorstore: {str(e)}. This may be due to issues with text extraction. Try another PDF with clear text content.")
+        st.toast(f"Error creating vectorstore. Try another PDF.", icon="⚠️")
```
st.toast as requested/suggested. The error should now be only one at a time and specific.

TheCarBun reviewed (changes requested):
this goes on a loop when I upload a PDF and the toast messages keep popping up! I think it broke 🥲
![image](https://github.com/user-attachments/assets/1ddfde21-9554-4377-9bc4-25921842d93c)

pakagronglb commented:
@TheCarBun OMG thats a long list of logs. That's odd. it works on my end for some reason. Let's try another solution. We'll figure it out. Don't worry ^^

pakagronglb commented:
@TheCarBun Ohh it was my mistake for doing st.rerun() too aggressively for error handling. I have to refractor those out.

pakagronglb commented:
So I tried doing that and the logs will now show this exact sequence happening once, without any
 repetition. The application is now working as intended, with proper error handling and no infinite loops.

TheCarBun commented:
@pakagronglb sure! Let's try that!!

pakagronglb commented:
So i added two modified files:
- app.py
- test_app.py

Created a commit with the message "fix: improve PDF processing and error handling, update tests"

Successfully pushed the changes to the fix/vector-store branch on the remote repository

The changes included:
- Improved error handling with specific and helpful messages
- Enhanced user experience by disabling text input during PDF processing failures
- Updated test cases to match the new implementation
- Fixed issues with PDF processing and vector store creation

now we pray

TheCarBun reviewed (approved):
LGTM🔥
